src/contracts/registry.ts

```diff
@@ -0,0 +1,68 @@
+import { getContractAddress } from "./contract";
+import { HexString } from "../types/Strings";
+import { getConfig } from "../config/config";
+import { ContractTransaction } from "ethers";
+import { MissingContract, MissingProvider, MissingSigner } from "../utilities";
+import { Registry__factory } from "../types/typechain";
+
+const CONTRACT_NAME = "Registry";
+
+/**
+ * Get the JSON RPC error from the body, if one exists
+ * @param e The error expected to have a vm Error
+ *
+ * @returns the error if any
+ */
+const getVmError = (e: { body?: string }): string | undefined => {
```

Comment:
I wonder if this should be extracted to a utility file somewhere. It seems like it could be used for other contract methods in the future.

Reply:
I thought about that as well, but I am hoping that we can switch to Solidity custom errors soon and it will not be so nasty. https://www.pivotaltracker.com/story/show/178337443

https://blog.soliditylang.org/2021/04/21/custom-errors/